### ISSUE TITLE
Error when using non-SFC vue component in storybook6

### ISSUE THREAD

**Issue by jackysee** [storybookjs/storybook#11960]
When using a non-SFC vue component in component meta. It shows error.

**To Reproduce**
Clone this repo 
https://github.com/jackysee/sb6-vue-non-sfc
and run storybook

**Screenshots**
![image](https://user-images.githubusercontent.com/30516/90089884-2622e080-dd55-11ea-84cc-98c62dfb35d8.png)

**Code snippets**
The component code `hello.js`
```
const comp = {
  name: "compA",
  render(h) {
    return h("div", "hello " + this.name);
  },
  props: {
    name: { type: String },
  },
};

export default comp;
```
The stories `hello.stories.js`
```
import hello from "./hello";

export default {
  title: "hello",
  component: hello,
};

export const basic = () => ({
  components: { hello },
  template: `<div><hello name="john"/></div>`,
});
```

The error:
```
TypeError: Cannot convert undefined or null to object
    at Function.entries (<anonymous>)
    at http://manjaro:6007/vendors~main.bdcd4fc7d8407f04ce17.bundle.js:9054:12
    at Array.reduce (<anonymous>)
    at combineParameters (http://manjaro:6007/vendors~main.bdcd4fc7d8407f04ce17.bundle.js:9053:32)
    at enhanceArgTypes (http://manjaro:6007/vendors~main.bdcd4fc7d8407f04ce17.bundle.js:4143:55)
    at _this$_argTypesEnhanc._argTypesEnhancers.reduce.Object.assign.__isArgsStory (http://manjaro:6007/vendors~main.bdcd4fc7d8407f04ce17.bundle.js:9882:21)
    at Array.reduce (<anonymous>)
    at StoryStore.addStory (http://manjaro:6007/vendors~main.bdcd4fc7d8407f04ce17.bundle.js:9880:59)
    at Object.api.add (http://manjaro:6007/vendors~main.bdcd4fc7d8407f04ce17.bundle.js:8343:25)
    at http://manjaro:6007/vendors~main.bdcd4fc7d8407f04ce17.bundle.js:22345:16
```

**System:**
The results of `npx sb@next info` 
```
Environment Info:

  System:
    OS: Linux 5.4 Manjaro Linux
    CPU: (2) x64 Intel(R) Core(TM) i5-5200U CPU @ 2.20GHz
  Binaries:
    Node: 12.13.0 - /tmp/fnm-shell-8468186/bin/node
    npm: 6.12.0 - /tmp/fnm-shell-8468186/bin/npm
  Browsers:
    Firefox: 79.0
  npmPackages:
    @storybook/addon-actions: ^6.0.5 => 6.0.5
    @storybook/addon-essentials: ^6.0.5 => 6.0.5
    @storybook/addon-links: ^6.0.5 => 6.0.5
    @storybook/vue: ^6.0.5 => 6.0.5

```

**Comment by piemasters** [comment]
I'm getting the same error having upgraded from storybook 6.0.0-beta.31 to 6.0.4

**Comment by jackysee** [comment]
The error is gone in 6.0.7. 
Except the it can't seems to load the props table, maybe  due to webpack setting on only reading `.vue` file?

**Comment by piemasters** [comment]
Yup 6.0.7 fixed it for me too! Props are working for me, but I'm using the args syntax (which is still in dev for vue) with the controls plugin. Try updating your story to this to see if it fixes it (it works nicely with the controls plugin if you have it):

```
import hello from "./hello";

export default {
  title: "hello",
  component: hello,
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { hello },
  template:
    '<div><hello :name="name"/></div>',
});

export const Default = Template.bind({});`
Default.args = {
  name: "john"
};

```

**Comment by shilman** [comment]
Aha fixed in #12012 (unintentionally!)
